### PR TITLE
Update ACME challenge link

### LINKED ISSUE #63: README link needs updated
Currently, we link to `https://letsencrypt.github.io/acme-spec/` in `README.rst`, but that displays the following when opened:

```
NOTE: This Repository is Deprecated

This repository is not active and does not accurately reflect what Let's Encrypt currently implements. It's retained only for history.

All new work happens here: https://github.com/ietf-wg-acme/acme/ 
```

The `README` file should be updated in reference to this notice. 

**Thank you to @michaelfrance for reporting this issue.**

### PR DESCRIPTION
**Pull request by paolodina**
Closes #63

### CODE REVIEW

**Comment by urda** [comment]
First of all big THANK YOU to you, @paolodina !

I've got a big week at work, but on the surface I don't see any issues with this. I'll make some time later this week to sit down and get this worked on 😄

File: README.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@ Let's Encrypt App for Django
    :target: https://codecov.io/gh/urda/django-letsencrypt/branch/master
 
 :code:`django-letsencrypt` will allow you to add, remove, and update any
-`ACME challenge <https://letsencrypt.github.io/acme-spec/>`_ objects you may
+`ACME challenge <https://tools.ietf.org/html/draft-ietf-acme-acme/>`_ objects you may
```
Comment (urda) [inline README.rst:11]: I would prefer if we linked to this URL that is listed at the old endpoint:

```
https://github.com/ietf-wg-acme/acme/
```

As it is quoted from:

```
NOTE: This Repository is Deprecated

This repository is not active and does not accurately reflect what Let's Encrypt currently implements. It's retained only for history.

All new work happens here: https://github.com/ietf-wg-acme/acme/ 
```

**Review by urda — CHANGES_REQUESTED** [review]
Hey can you make these changes and re-base your commit please?

File: README.rst (hunk 1 above)
Comment (urda) [reply README.rst:11]: You will also need to rebase on to the `develop` branch for me.

**Comment by urda** [comment]
Closing in favor of #65 , please be sure next time to update the same branch in the PR when you rebase or update after a review. I just happened to notice your proper commit in your `develop` branch before I went and cut my own commit.

**Comment by paolodina** [comment]
@urda I didn't mean to cause that trouble. Apologies.

**Comment by urda** [comment]
@paolodina No worries! I wanted to make sure your work got in! Contributions are ALWAYS welcome for review!

I pulled your commit, and I'm working on adding it to `develop`. 😄

**Pull request closed without merging**